packages/@uppy/companion/src/server/helpers/request.js

```diff
@@ -79,15 +79,23 @@ function isPrivateIP (ipAddress) {
 
 module.exports.FORBIDDEN_IP_ADDRESS = FORBIDDEN_IP_ADDRESS
 
-module.exports.getRedirectEvaluator = (requestURL, blockPrivateIPs) => {
-  const { protocol } = new URL(requestURL)
+module.exports.getRedirectEvaluator = (rawRequestURL, blockPrivateIPs) => {
+  const requestURL = new URL(rawRequestURL)
   return (res) => {
     if (!blockPrivateIPs) {
       return true
     }
 
     const redirectURL = res.headers.location
-    const shouldRedirect = redirectURL ? new URL(redirectURL).protocol === protocol : false
+    let shouldRedirect = false
+    if (redirectURL) {
+      try {
+        shouldRedirect = new URL(redirectURL, requestURL).protocol === requestURL.protocol
+      } catch (e) {
+        // `redirectURL` is invalid
+      }
+    }
+
     if (!shouldRedirect) {
```

Comment:
is this also logged when `!res.headers.location`? Then it wouldn't technically be blocking a redirect, but not having one to perform in the first place, right?

Reply:
The function is only called by the `request` package if the response is a redirect, so it should be fine like this

Reply:
Although the change for your other comment addresses this anyway 😇